docs/intro-react-native-components.md

```diff
@@ -0,0 +1,100 @@
+---
+id: intro-react-native-components
+title: Core Components and Native Components
+---
+
+React Native is an open source framework for building Android and iOS applications using [React](https://reactjs.org/) and the app platform’s native capabilities. With React Native, you use JavaScript to access your platform’s APIs as well as to describe the appearance and behaviour of your UI using React components, bundles of reusable, nestable code. You can learn more about React in the next section. But first, let's cover how components work in React Native.
+
+## Views and mobile development
+
+In Android and iOS development, a **view** is the basic building block of UI: a small rectangular element on the screen which can be used to display text, images, or respond to user input. Even the smallest visual elements of an app, like a line of text or a button, are kinds of views. Some kinds of views can nest other views. It’s views all the way down!
+
+## Native Components
+
+In Android development, you write views in Java or Kotlin; in iOS development, you use Objective-C or Swift. With React Native, you can invoke these views with JavaScript using React components. At runtime, React Native creates the corresponding Android and iOS views for those components. Because React Native components are backed by the same views as Android and iOS, React Native apps look, feel, and perform like any other apps. We call these platform-backed components Native Components.
```

Comment:
Given the recent popularity of newer languages, I'd reverse the order: "Kotlin or Java; in iOS development you use Swift or Objective-C"

Also may be nice to make "Native Components" at the end bold so that it connects with the headline. You are driving the point home so nicely about what they are but by the end of the paragraph I kinda forgot that was the focus of the paragraph (might just be because your writing is that good I got lost!).

Reply:
Yeah, I got confused that "... Native Components" was immediately followed by "Core Components" and thought there's a terminology mistake. I didn't noticed that this section had a "Native Components" headline.

One possible fix would be to introduce the name in the beginning of this section.

For example:

>## Native Components

>Some components are provided by the platform. We call these platform-backed components *Native Components*. For example, in Android [...]

Reply:
It should be bolded because each new Important Term needs a bolding, like in school texts. Good catch! I missed it!

Reply:
Dan, I worry that the phrasing you suggest pops right in with "components" before introducing the concept of components. We were _just_ talking about views and how both mobile platforms use them for everything in their apps and then suddenly "platforms provide components! We call them..." Wait. What. Is. A. Component. I worry we're skipping ahead a bit if we frontload with that. This is a common problem I noticed with folks unfamiliar with either views or components: gotta build a bridge to link the two explicitly.

Reply:
I trust your choice here

Reply:
Went with Christoph's edits. Let's watch learners in the wild and see if this gels with them!